src/listeners.js

```diff
@@ -0,0 +1,180 @@
+    );
+    debug('Listening to hook exchanges');
+    this.pulseHookChangedListener = consumer;
+    this.listeners = [];
+  }
+
+  /** Create a new pulse consumer for a hook */
+  async createListener({hook, oldBindings}) {
+    this.hook = hook;
+    const client =  this.client;
+    const queueName = `${hook.hookGroupId}/${hook.hookId}`; // serves as unique id for every listener
+    const listener = await pulse.consume({
+      client,      
+      queueName: queueName, 
+      maxLength : 50,
+    }, async ({payload}) => {
+      const hook = this.hook;
+      // Pass empty functions in res which are executed by `triggerHookCommon`
+      // This is more of a hack so that if the payload doesnt validate against
+      // `triggerSchema`,git  we can simply ignore the message.
+      await this.taskcreator.fire(hook, {firedBy:'pulseMessage', payload});
+    });
+    if (!this.client.isFakeClient) {
+      for (let {exchange, routingKeyPattern} of oldBindings) {
+        await this.client.withChannel(async channel => channel.unbindQueue(queueName, exchange, routingKeyPattern));
+      }
+      for (let {exchange, routingKeyPattern} of hook.bindings) {
+        await this.client.withChannel(async channel => {channel.bindQueue(queueName, exchange, routingKeyPattern);});
+      }
+    }
+    this.listeners.push(listener);
+  }
+  
+  async reconcileConsumers() {
+    let Queues = [];
+    let allHooksData = [];
+    await this.Queues.scan(
+      {},
+      {
+        limit: 1000,
+        handler:(queue) => Queues.push(queue),
+      }
+    );
+    let continuationToken = undefined;
+    
+    do {
+      allHooksData  = await this.Hook.scan({}, {
+        limit: 1000,
+        continuation: continuationToken,
+      });
+      continuationToken = allHooksData.continuation;
+      allHooksData.entries.forEach(async (hook) => {
+        if (hook.bindings != []) {
+          const {hookGroupId, hookId} = hook;
+          const queue = _.find(Queues, {hookGroupId, hookId});
+          if (queue) {
+            const index = this.listeners.findIndex(({_queueName}) => listener._queueName === queue.queueName);
+            if (index == -1) {
+              debug('Existing queue..creating listener');
+              await this.createListener({hook, oldBindings: queue.bindings});
+            }
+            _.pull(Queues, queue);
+          } else {
+            debug('New queue..creating listener');
+            await this.createListener({hook, oldBindings: [{}]});
+            // Add to Queues table
+            debug('adding to Queues table');
+            await this.Queues.create({
+              hookGroupId,
+              hookId,
+              queueName: `${hookGroupId}/${hookId}`,
+              bindings: hook.bindings,
+            });
+          }
+        }
+      });
+    } while (continuationToken);
+    
+    // Delete the queues now left in the Queues list.
+    Queues.forEach(async (queue) => {     
```

Comment:
For parity with `createListener`, it probably makes sense to put this in a function named `removeQueue`.